deploy.sh

```diff
@@ -0,0 +1,233 @@
+cleanUpDockerImages(){
+    set +e
+    local to_be_removed="$(echo $SUDO_PASSWORD |sudo -S docker images ${DOCKER_NAME} -q)  \
+                         $(echo $SUDO_PASSWORD |sudo -S docker images pipeline/image-service-base -q) \
+                         $(echo $SUDO_PASSWORD |sudo -S docker images -f "dangling=true" -q )"
+    # remove ${DOCKER_NAME} image,  pipeline/image-service-base and <none>:<none> images
+    if [ ! -z "${to_be_removed// }" ] ; then
+         echo $SUDO_PASSWORD |sudo -S docker rmi $to_be_removed
+    fi
+    set -e
+}
+
+##############################################
+#
+# Remove docker instance which are running
+#
+###########################################
+cleanUpDockerContainer(){
+    set +e
+    local docker_name_key=$1
+    local running_docker=$(echo $SUDO_PASSWORD |sudo -S docker ps -a |grep "$1" |awk '{print $1}')
+    if [ "$running_docker" != "" ]; then
+         echo $SUDO_PASSWORD |sudo -S docker stop $running_docker
+         echo $SUDO_PASSWORD |sudo -S docker rm   $running_docker
+    fi
+    set -e
+}
+
+######################################
+#
+# Clean Up runnning docker instance
+#
+#####################################
+cleanupDockers(){
+    echo "CleanUp Dockers ..."
+    set +e
+    cleanUpDockerContainer "${DOCKER_NAME}"
+    cleanUpDockerImages
+    set -e
+}
+
+############################################
+#
+# Clean Up if you want to stop image service docker and recover services
+#
+###########################################
+cleanUp(){
+    set +e
+    echo "*****************************************************************************************************"
+    echo "Start to clean up environment: stopping running containers, starting service mongodb and rabbitmq-server"
+    echo "*****************************************************************************************************"
+    cleanupDockers
+    netstat -ntlp
+    echo "*****************************************************************************************************"
+    echo "End to clean up environment: stopping running containers, starting service mongodb and rabbitmq-server"
+    echo "*****************************************************************************************************"
+    set -e
+}
+
+###################################
+#
+# Build docker and run it
+#
+#################################
+dockerUp(){
+    echo "*****************************************************************************************************"
+    echo "Start to build and run image service docker"
+    echo "*****************************************************************************************************"
+    mkdir -p  $WORKSPACE/build-logs
+    pushd $REPO_DIR
+    echo $SUDO_PASSWORD |sudo -S docker build -t $DOCKER_NAME .
+    echo $SUDO_PASSWORD |sudo -S docker run --privileged --net=host -v /etc/localtime:/etc/localtime:ro -t $DOCKER_NAME > $WORKSPACE/build-logs/image-service.log &
```

Comment:
how about we start docker with "--rm" parameter, so that it will be self-removed when ```docker stop```
to avoid problem manuallly "docker rm" ?